### PR TITLE
Add/805 playlist refactor

### PR DESCRIPTION
**Pull request by antpb**
## Description
This is a clean PR to add the playlist block to Gutenberg. There were merge conflicts that resulted in me needing to create a new branch based on the newest version of Gutenberg. Previous conversation can be found here: https://github.com/WordPress/gutenberg/pull/7126

CC: @karmatosed @noisysocks

### CODE REVIEW

**Comment by karmatosed** [comment]
Thanks for pushing on this @antpb, looks unfortunately like we have merge conflicts again. Any chance of trying to clean those?

File: core-blocks/playlist/edit.js [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+		const { setAttributes } = this.props;
+		editorMediaUpload( {
+			allowedType: [ 'audio', 'video' ],
+			filesList: files,
+			onFileChange: ( media ) => {
+				const firstType = get( files, [ 0, 'mimeType' ] );
+				const isConsistentType = !! firstType && every( files, ( filesMedia ) => filesMedia.mimeType === firstType );
+				//validate type is consistent for playlist
+				if ( ! isConsistentType ) {
+					console.log( 'I cannot let you do that. Error notice will go here.' );
```
[the remark this answers is not in the document]
Comment (noisysocks) [reply core-blocks/playlist/edit.js]: It's unfortunately not documented but you could wrap the component with the `withNotices` HOC to do this. The File block has an example on how it's used. The basic idea is that you feed `this.props.noticeUI` into `<MediaPlaceholder>` and then call `noticeOperations.createErrorNotice( message )` whenever you wish to show an error notice.

File: core-blocks/playlist/index.php [hunk 2]
```diff
@@ -34,34 +34,34 @@ function register_block_core_playlist() {
...
 			),
 			'style'            => array(
-				'type'    => 'string',
-				'default' => 'light',
+						'type'    => 'string',
+						'default' => 'light',
 			),
 			'align'            => array(
-				'type' => 'string',
+						'type' => 'string',
```
Comment (antpb) [inline core-blocks/playlist/index.php]: @ianbelanger79 does the => need to move over on this line a bit?

Comment (ianbelanger79) [reply core-blocks/playlist/index.php]: Yes it should for readability, good catch @antpb, I'll submit the change

**Pull request closed without merging**